source/parser/util.civet

```diff
@@ -569,22 +570,46 @@ function wrapIIFE(expressions: StatementTuple[], asyncFlag?: boolean): ASTNode
   signature :=
     modifier:
       async: !!async.length
+      generator: !!generator
     returnType: undefined
 
-  fn := makeNode {
-    type: "ArrowFunction"
-    signature
-    parameters
-    returnType: undefined
-    ts: false
-    async
-    block
-    children: [async, parameters, "=>", block]
-  }
+  let fn
+  if generator
+    fn = makeNode {
+      type: "FunctionExpression"
+      signature
+      parameters
+      returnType: undefined
+      ts: false
+      async
+      block
+      generator
+      children: [async, "function", generator, parameters, block]
+    }
+  else
+    fn = makeNode {
+      type: "ArrowFunction"
+      signature
+      parameters
+      returnType: undefined
+      ts: false
+      async
+      block
+      children: [async, parameters, "=>", block]
+    }
 
-  exp := makeNode
+  children := [ makeLeftHandSideExpression(fn), "()" ]
+
+  if fn.type is "FunctionExpression"
+    if gatherRecursiveWithinFunction(block, (is like { token: "this" }))#
+      children.splice 1, 0, ".bind(this)"
+    if gatherRecursiveWithinFunction(block, (is like { token: "arguments" }))#
```

Comment:
In the future we'll want to check to see if this `arguments` is declared in a surrounding scope or is the implicit function `arguments` but this is good for now. I don't think our variable scoping machinery makes this very easy yet.

Reply:
Oh, I didn't realize `arguments` could be bound to. Fun! Agreed, this should be doable, but probably best left for future work.